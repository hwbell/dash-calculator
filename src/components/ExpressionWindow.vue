<template>
  <div class="expression-window">
    <div class="expression-text" :class="alert && 'alert-text'">{{expression}}&nbsp;</div>
    <div class="answer-text" v-if="answer">{{` = ${answer}`}}</div>   

  </div>
</template>

<script>
export default {
  name: "Calculator",
  props: {
    expression: String,
    answer: Number,
    alert: Boolean
  },

 
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

.expression-window {
  height: 60px;
  width: 400px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-end;
  border-radius: 4px;
  border: 2px solid grey;
}
.answer-text {
  color: blue;
  font-weight: bolder;
}

.expression-text {
  transition: all 0.3s ease;
}
.alert-text {
  /* font-weight: bolder; */
  transform: translateY(-6px);
  color: red; 
}

/* h3 {
  margin: 40px 0 0;
}
.calculator {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.calculator-buttons {
  width: 80%;
}
.calculator-button {
  padding: 10px;
  width: 100px;
  border-radius: 0;
  border: 1px solid rgb(255, 255, 255); */
/* } */
</style>
