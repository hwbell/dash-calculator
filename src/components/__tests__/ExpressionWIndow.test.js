import { mount } from '@vue/test-utils'
import ExpressionWindow from '../ExpressionWindow.vue'

describe('ExpressionWindow', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(ExpressionWindow, {
      propsData: {
        expression: '4*3-2'
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.props().expression).toBe('4*3-2');
  })
  
  test('renders a button for each object in the array map', () => {
    // test a unique button(s)
    let window = wrapper.find('.expression-window');
    expect(window).toBeTruthy();
  })

})