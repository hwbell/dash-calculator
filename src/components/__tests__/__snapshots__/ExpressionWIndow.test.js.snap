// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpressionWindow renders correctly 1`] = `
<div
  class="expression-window"
>
  <div
    class="expression-text"
  >
    4*3-2Â 
  </div>
   
  <!---->
</div>
`;
